src/Uplink/Admin/Fields/Field.php

```diff
@@ -49,21 +47,47 @@ class Field {
 	protected bool $show_heading = false;
 
 	/**
-	 * Constructor!
-	 *
-	 * @param string $slug Field slug.
+	 * @var string
 	 */
-	public function __construct( string $slug ) {
-		$this->slug = $slug;
+	protected const VIEW = 'admin/fields/field';
 
-		$collection = Config::get_container()->get( Collection::class );
-		$resource   = $collection->get( $slug );
+	/**
+	 * @var View
+	 */
+	protected $view;
 
-		if ( ! $resource instanceof Resource ) {
-			throw new \InvalidArgumentException( sprintf( 'Resource with slug "%s" does not exist.', $slug ) );
-		}
+	/**
+	 * @var Assets
+	 */
+	protected $assets;
+
+	/**
+	 * @var Group
+	 */
+	protected $group;
 
+	/**
+	 * Constructor!
+	 *
+	 * @param  View  $view  The View Engine to render views.
+	 */
+	public function __construct( View $view, Assets $assets, Group $group ) {
+		$this->view   = $view;
+		$this->assets = $assets;
+		$this->group  = $group;
+	}
+
+	/**
+	 * Sets the resource.
+	 *
+	 * @param  Resource  $resource  The resource.
+	 *
+	 * @return self
```

Comment:
In case this gets extended.

```suggestion
	 * @return static
```